mlrun/runtimes/pod.py

```diff
@@ -412,6 +413,26 @@ def _verify_and_set_limits(
                         resource_value,
                     )
 
+    def add_volumes(self, volumes):
```

Comment:
A few comments here since this is now user facing:
1. Add type hints
2. Move above `_set_volume_mount` to maintain reading order
3. Suggest to rename - we usually use `set_...` or `with_...` I think set is more fitting and that we should keep some conventions here to make it easier to find.
4. Expand the docstring - IIUC users shouldn't really be using this and this is mainly for `auto_mount` usage